zebedee/collection.go

```diff
@@ -113,3 +115,38 @@ func (z *zebedeeClient) UpdateCollection(s Session, desc CollectionDescription)
 
 	return z.executeRequestNoResponse(req, http.StatusOK)
 }
+
+//UpdateCollectionContent updates content within a collection
+//  overwriteExisting (default:true) - if set to false, any existing content will not be overwritten
+//  recursive (default:false) - if set to true, all associated files alongside the page will be added to the collection's in progress directory
+//              if set to false, only the data.json file will be added to the collection's in progress directory
+//  validateJson (default:true) - if set to true, the json will be validated to ensure it's a valid page JSON structure
+func (z *zebedeeClient) UpdateCollectionContent(
+	s Session,
+	id, contentUri string,
+	content io.Reader,
```

Comment:
1.  Does this need to be an `io.Reader`? I know the page/content types that are written to this endpoint are varied so a single type won't work but could we use `interface{}` instead?  I think that would offer the same flexibility but also allow you to reuse the `z.newAuthenticatedRequest()` func to create the requests. 

2. What are the overwriteExisting, recursive and validate JSON defaults? If there is a consistent default could we apply that internally within this method and keep the interface simple by reducing the number of params..

I'm aware I am nit picking but I want to keep it clean and simple where possible.

Reply:
Fair points
> 1. Does this need to be an `io.Reader`? I know the page/content types that are written to this endpoint are varied so a single type won't work but could we use `interface{}` instead?  I think that would offer the same flexibility but also allow you to reuse the `z.newAuthenticatedRequest()` func to create the requests.

In the case of the collection API I don't think the JSON content will be unmarshalled before calling Zebedee - it will just be passed through the API. This is the reasoning behind having the reader, as it can pass the reader from the body of the request without having to unmarshall. 🤔 

> 2. What are the overwriteExisting, recursive and validate JSON defaults? If there is a consistent default could we apply that internally within this method and keep the interface simple by reducing the number of params..

I added some comments to the function header describing their purpose. I debated what to do with these - they are a side effect of the content endpoint doing too much. I had considered setting defaults, though Florence will use different values depending on the content type. I get your point about the params being too much - I guess another option would be to use this function within the package, and expose multiple content function for the different variations of the parameters. Trouble is it's not easy to decipher what they are 🤯

Reply:
I have reworked the interface to
 - use the `interface{}` instead of `io.Reader`. It's seems likely that the JSON will be unmarshalled to some extent to it seems the right solution 👍 
 - I have applied the default values for these flags internally. If there are variations required, then the decision can be made how to expose the variations.